src/Microsoft.ApplicationInsights/TelemetryClient.cs

```diff
@@ -513,44 +510,94 @@ public void Initialize(ITelemetry telemetry)
+                                                            "Exception while initializing {0}, exception message - {1}",
+                                                            this.configuration.TelemetryInitializers[index].GetType().FullName,
+                                                            exception));
+                        }
+                    }
                 }
             }
 
-            if (telemetry.Timestamp == default(DateTimeOffset))
+            if (!sampledOut)
             {
-                telemetry.Timestamp = PreciseTimestamp.GetUtcNow();
-            }
+                if (telemetry.Timestamp == default(DateTimeOffset))
+                {
+                    telemetry.Timestamp = PreciseTimestamp.GetUtcNow();
+                }
 
-            // Currently backend requires SDK version to comply "name: version"
-            if (string.IsNullOrEmpty(telemetry.Context.Internal.SdkVersion))
-            {
-                var version = this.sdkVersion ?? (this.sdkVersion = SdkVersionUtils.GetSdkVersion(VersionPrefix));
-                telemetry.Context.Internal.SdkVersion = version;
-            }
+                // Currently backend requires SDK version to comply "name: version"
+                if (string.IsNullOrEmpty(telemetry.Context.Internal.SdkVersion))
+                {
+                    var version = this.sdkVersion ?? (this.sdkVersion = SdkVersionUtils.GetSdkVersion(VersionPrefix));
+                    telemetry.Context.Internal.SdkVersion = version;
+                }
 
-            // set NodeName to the machine name if it's not initialized yet, if RoleInstance is also not set then we send only RoleInstance
-            if (string.IsNullOrEmpty(telemetry.Context.Internal.NodeName) && !string.IsNullOrEmpty(telemetry.Context.Cloud.RoleInstance))
-            {
-                telemetry.Context.Internal.NodeName = PlatformSingleton.Current.GetMachineName();
-            }
+                // set NodeName to the machine name if it's not initialized yet, if RoleInstance is also not set then we send only RoleInstance
+                if (string.IsNullOrEmpty(telemetry.Context.Internal.NodeName) && !string.IsNullOrEmpty(telemetry.Context.Cloud.RoleInstance))
+                {
+                    telemetry.Context.Internal.NodeName = PlatformSingleton.Current.GetMachineName();
+                }
 
-            // set RoleInstance to the machine name if it's not initialized yet
-            if (string.IsNullOrEmpty(telemetry.Context.Cloud.RoleInstance))
-            {
-                telemetry.Context.Cloud.RoleInstance = PlatformSingleton.Current.GetMachineName();
+                // set RoleInstance to the machine name if it's not initialized yet
+                if (string.IsNullOrEmpty(telemetry.Context.Cloud.RoleInstance))
+                {
+                    telemetry.Context.Cloud.RoleInstance = PlatformSingleton.Current.GetMachineName();
+                }
             }
```

Comment:
You're sitting next to me right now and told me that we need to review QuickPulse because these might be required.
Adding comment here so we don't forget :)

Reply:
Yep, manual test still remains. Upon code investigation, it looks like we'll need to use "deferRequestProperties" flag in addition to "proactiveSampling" flag in .NET Core App Services to compensate for the absence of URL on sampled requests just like you do in dotnet-server repo.